core/src/main/java/tc/oc/pgm/core/Core.java

```diff
@@ -161,7 +162,7 @@ public void markLeaked() {
     this.leaked = true;
   }
 
-  public List<String> getModeList() {
+  public List<Mode> getModeList() {
```

Comment:
```suggestion
  public Set<Mode> getModeList() {
```
Across the code, replace List with Set, you're primarily going to be doing `contains` on this collection, sets are better for that

Reply:
Is there a reason why these shouldn't be ImmutableSets? These are after the sets are created and passed to the core object.